tapalcatl_server/handler.go

```diff
@@ -25,23 +28,24 @@ type BufferManager interface {
 	Put(*bytes.Buffer)
 }
 
-// the logState structures and string generation serve to emit a single log entry line
+// the reqState structures and string generation serve to emit a single log entry line
 // a log parser will pick this up and use it to persist metrics
 // the string functions here are specific to the format used and should be updated with care
 
-type logResponseState int32
+type reqResponseState int32
 
 const (
-	ResponseState_Nil logResponseState = iota
+	ResponseState_Nil reqResponseState = iota
```

Comment:
> I'm not sure. I think it is considered good practice to have a "zero state" to be the default initialised value, but I'm not sure how that works when that value isn't meaningful. What does it mean if we get to the end of MetatileHandler's HandlerFunc and we haven't set the value to something other than ResponseState_Nil?

Yea, that's why I made the meaninglessness explicit :) It seemed better than having the first value default to either error or success as a code error and not be able to tell downstream.

> Ideally, we could have the compiler check that no such code path exists. But I can't think of a way to do that, so perhaps we should log ResponseState_Nil in the defer block as an error?

Another approach is to only allow the creation of these objects through a function, which takes the state explicitly. But for the compiler to enforce that we would need to shuffle packages around. I actually considered that anyway just for organizational reasons, but I think it's nice for the request processing state to sit close to the handler.

I logged an error in 0ba6a5e.